native-schema-registry/python/PyGsrSerDe/KafkaSerDe.py

```diff
@@ -1,43 +1,87 @@
 from .PyGsrSerDe import *
 from .AvroSerDe import *
+from .GlueSchemaRegistryConfiguration import *
+
 
 class GlueSchemaRegistryKafkaSerializer:
-    def __init__(self):  # data format design needs discussion
-        self.data_format = "AVRO"
+    """A serializer to process and serialize kafka records."""
+
+    def __init__(self, configs: dict):
+        """Create a GlueSchemaRegistryKafkaSerializer object."""
+        self.config = GlueSchemaRegistryConfiguration(configs)
         self.gsr_serializer = GlueSchemaRegistrySerializer
         serializers = {
-            "AVRO": AvroSerializer(),
-            "JSON": "JSONSerializer()",  # placeholders
-            "PROTOBUF": "ProtobufSerializer()",
+            "AVRO": self.getAvroSerializer(),
+            "JSON": "self.getJSONSerializer()",  # placeholders
+            "PROTOBUF": "self.getProtobufSerializer()",
         }
-        self.data_format_serializer = serializers.get(self.data_format)
+        self.data_format_serializer = serializers.get(self.config.dataformat)
+
+    def getAvroSerializer(self):
+        """Get method for AvroSerializer. Prevents having to initialize unused serializers."""
+        return AvroSerializer()
 
-    def serialize(self, data: dict, schema: dict, topic: str):
+    def serialize_for_kafka_python(self, data: dict):
+        """Process and serialize a record."""
         if data is None:
             return None
-        if schema is None:
+
+        serialized_record = self.data_format_serializer.serialize(data, self.config.schema)
+
+        schema_name = self.config.schema_naming_strategy.getSchemaName(data, self.config.topic)
+        gsr_schema = GlueSchemaRegistrySchema(schema_name, self.config.schema, self.config.dataformat)
+
+        encoded = self.gsr_serializer.encode(self.config.topic, gsr_schema, serialized_record)
+
+        return encoded
+
+    def serialize_for_confluent(self, data: dict, serializationcontext: object = None):
+        """Process and serialize a record, with serializationcontext as an additional input."""
+        if data is None:
             return None
 
-        serialized_record = self.data_format_serializer.serialize(data, schema)
+        serialized_record = self.data_format_serializer.serialize(data, self.config.schema)
 
-        gsr_schema = GlueSchemaRegistrySchema("schema_name", schema, self.data_format)
+        schema_name = self.config.schema_naming_strategy.getSchemaName(data, serializationcontext.topic)
+        gsr_schema = GlueSchemaRegistrySchema(schema_name, self.config.schema, self.config.dataformat)
 
-        encoded = self.gsr_serializer.encode(topic, gsr_schema, serialized_record)
+        encoded = self.gsr_serializer.encode(serializationcontext.topic, gsr_schema, serialized_record)
```

Comment:
Any suggestions on what to extract here? There already isn't much logic here, extracting the lines here feel like it would just create another nested layer without saving much.